register/register.go

```diff
@@ -34,6 +34,12 @@ type SLSAVerifier interface {
 		provenanceOpts *options.ProvenanceOpts,
 		builderOpts *options.BuilderOpts,
 	) ([]byte, *utils.TrustedBuilderID, error)
+
+	VerifyNpmPackageWithSigstoreTUFClient(ctx context.Context,
```

Comment:
As we chatted about offline, we may want this API to take the npm verification material directly and another API responsible for fetching this target file from TUF. Then this API could verify metadata offline, and the latter API would be responsible for network calls. I think this would also make testing easier since you don't have to mock the Sigstore TUF client, just need the verification material struct initialized.

Another change, outside of the scope of this PR, would be to update https://github.com/slsa-framework/slsa-verifier/blob/main/verifiers/internal/gha/trusted_root.go to use [TrustedMaterial](https://github.com/sigstore/sigstore-go/blob/main/pkg/root/trusted_material.go) rather than Cosign's APIs. Like you're already doing in https://github.com/slsa-framework/slsa-verifier/blob/main/verifiers/internal/gha/bundle.go#L162, rather than passing the TUF client, you would pass in the trusted root material.

Reply:
Speaking about this offline again, we have our own struct to parse the material, but the spec/layout of the material isn't defined in npmjs docs. Until we can be sure that npmjs will commit to this layout, I think it's better to hide this detail from the user, or let the user stub out their own `SigstoreTUFClient`.
 - https://docs.npmjs.com/about-registry-signatures#supporting-signatures-on-third-party-registries

Reply:
I don't think we should recommend implementing a `SigstoreTUFClient` to bypass TUF. I think this API is fine for the current use case, though taking in the verification material (keys) directly could be another way to implement this API while not requiring knowledge of the npm struct.

Reply:
in the long term this function should use variadic options / arguments. For a quick turnaround, maybe you could add a pointer to a tuf client in the existing builderOpts https://github.com/slsa-framework/slsa-verifier/blob/main/options/options.go#L36.

Reply:
If you're referring to using variadic options like in this blog post, I decided on a bit of a hybrid approach in my latest commits.
- https://petomalina.medium.com/dealing-with-optional-parameters-in-go-9780f9bfbd1d